src/sql/workbench/browser/modelComponents/componentBase.ts

```diff
@@ -299,7 +299,9 @@ export abstract class ContainerBase<T, TPropertyBag extends azdata.ComponentProp
 
 	/// IComponent container-related implementation
 	public addToContainer(componentDescriptor: IComponentDescriptor, config: any, index?: number): void {
-		this.logService.debug(`Adding component ${componentDescriptor.id} to container ${this.descriptor.id}`);
+		if (componentDescriptor && this.descriptor) {
```

Comment:
Hmmm, I'd like to know why this is being called when the descriptors aren't set. Were you seeing both of these be undefined or just this.descriptor?

Every component needs to have a descriptor at some point - the issue is that it's injected after the component is constructed so if something is happening in the constructor that causes it to start calling add/remove before that happens then this could break. 

Can you include the callstack at the point where you hit this and one of these is undefined?

Additionally if you can provide a repro that I can do myself I'll look into it.